### PR TITLE
support on demand resources

### PR DESCRIPTION
**Pull request by JunyiXie**
depand on 
Core: https://github.com/CocoaPods/Core/pull/622
Xcodeproj: https://github.com/CocoaPods/Xcodeproj/pull/742

### CODE REVIEW

**Comment by dnkoutso** [comment]
@JunyiXie still want to try to land this? Needs a rebase.

**Comment by cillyfly** [comment]
@JunyiXie  I have check out and run it but when I used to run  ../../bin/pod update but an error can out 

Invalid `TestLibrary.podspec` file: undefined method `on_demand_resources=' for #<Pod::Specification name="TestLibrary">.

I have been using gem build cocoapods.gemspec and gem install cocoapods-1.9.1.gem

is that something wrong?

File: lib/cocoapods/installer/user_project_integrator/target_integrator.rb [hunk 1]
```diff
@@ -418,6 +419,78 @@ def inspect
...
+          # Clean up previous on_demand_resources group
+          old_on_demand_resource_files = {}
+          on_demand_resources_group.groups.each do |group|
+            old_on_demand_resource_files[group.name] = []
+
+            group.files.each {|file|
+              old_on_demand_resource_files[group.name] << file
+            }
+          end
+          pb_target = target.user_project.objects_by_uuid[target.user_target_uuids[0]]
```
Comment (cillyfly) [inline lib/cocoapods/installer/user_project_integrator/target_integrator.rb:455]: `target.user_target_uuids[0]` is this means that `pb_target` always will be the first target? 

is that support mult targets？

just like
```
target 'TestLibrary_Tests' do
  pod 'TestLibrary', :path => '../'

  target 'TestLibrary' do  
end 
end

```

**Comment by dnkoutso** [comment]
Superseded by https://github.com/CocoaPods/CocoaPods/pull/10812

**Pull request closed without merging**